src/compiler/program.ts

```diff
@@ -1015,7 +1015,7 @@ namespace ts {
             let declarationDiagnostics: Diagnostic[] = [];
 
             if (options.noEmit) {

```

Comment:
Why are there two places where we check `noEmit`? One in `program.ts` and one in `emitter.ts`?